install/kubernetes/helm/istio/charts/gateways/templates/deployment.yaml

```diff
@@ -164,6 +164,10 @@ spec:
             periodSeconds: 2
             successThreshold: 1
             timeoutSeconds: 1
+          lifecycle:
+            preStop:
+              exec:
+                command: ["/bin/bash", "-c", "sleep 30"]
```

Comment:
I am not sure how can this make envoy gracefully shutdown.

xref: https://kubernetes.io/docs/concepts/workloads/pods/pod/#termination-of-pods

Reply:
From my understanding, this will "drain" Envoy since the endpoints are being removed and after certain time, no requests is going to the terminating Envoy pod.

Reply:
I know what you mean.  From this point, it is meaningful.

Reply:
This won't drain the local Envoy. It gives Pilot 30s to tell the other Envoys that this pod is going to terminate so remove it from your endpoint pool. Ongoing connections will be handled by the new draining functionality we added in 1.1.0.

Reply:
> This won't drain the local Envoy. It gives Pilot 30s to tell the other Envoys that this pod is going to terminate so remove it from your endpoint pool. Ongoing connections will be handled by the new draining functionality we added in 1.1.0.

That's reason why I add the double quote to drain. :P